hera_mc/cm_sysutils.py

```diff
@@ -424,3 +424,238 @@ def get_apriori_antenna_status_for_rtp(self, status, at_date='now'):
+    """
+    hu = cm_hookup.Hookup(session)
+    na_from_file = node_antennas('file', session=session)
+    na_from_hookup = node_antennas(hu, session=session)
+    info = {'nodes': []}
+    for node in node_num:
+        # Set up
+        if isinstance(node, int):
+            node = 'N{:02d}'.format(node)
+        npk = cm_utils.make_part_key(node, 'A')
+        info['nodes'].append(node)
+        info[node] = {}
+        # Get antenna info
+        try:
+            info[node]['ants-file'] = na_from_file[node]
+        except KeyError:
+            info[node]['ants-file'] = []
+        try:
+            info[node]['ants-hookup'] = na_from_hookup[node]
+        except KeyError:
+            info[node]['ants-hookup'] = []
+        # Get hookup info
+        snaps = hu.get_hookup(node, hookup_type='parts_hera')
+        wr = hu.get_hookup(node, hookup_type='wr_hera')
+        rd = hu.get_hookup(node, hookup_type='arduino_hera')
+        # Find snaps
+        info[node]['snaps'] = [cm_utils.split_part_key(x)[0] for x in snaps.keys()]
+        # Find white rabbit, arduino and node control module
+        wr_ret = _get_dict_elements(npk, wr, 'wr', 'ncm')
+        info[node]['wr'] = wr_ret['wr']
+        rd_ret = _get_dict_elements(npk, rd, 'rd', 'ncm')
+        info[node]['arduino'] = rd_ret['rd']
+        info[node]['ncm'] = ''
+        if len(wr_ret['ncm']) and len(rd_ret['ncm']) \
+                and wr_ret['ncm'] != rd_ret['ncm']:  # pragma: no cover
+            raise ValueError("NCMs don't match for node {}:  {} vs {}"
+                             .format(node, wr_ret['ncm'], rd_ret['ncm']))
+        elif len(wr_ret['ncm']):
+            info[node]['ncm'] = wr_ret['ncm']
+        elif len(rd_ret['ncm']):  # pragma: no cover
+            info[node]['ncm'] = rd_ret['ncm']
+
+        # Get notes
+        notes = hu.get_notes(snaps, state='all')
+        for snp in info[node]['snaps']:
+            spk = cm_utils.make_part_key(snp, 'A')
+            try:
+                info[snp] = list(notes[spk][spk].values())
+            except KeyError:
+                info[snp] = []
+        notes = hu.get_notes(wr, state='all')
+        wpk = cm_utils.make_part_key(info[node]['wr'], 'A')
+        try:
+            info[info[node]['wr']] = list(notes[npk][wpk].values())
+        except KeyError:
+            info[info[node]['wr']] = []
+        notes = hu.get_notes(rd, state='all')
+        apk = cm_utils.make_part_key(info[node]['arduino'], 'A')
+        try:
+            info[info[node]['arduino']] = list(notes[npk][apk].values())
+        except KeyError:
+            info[info[node]['arduino']] = []
+        if '' in info.keys():
+            del info['']
+    return info
+
+
+def _get_macip(info):
+    data = []
+    for this_note in info:
+        if this_note.startswith('MAC'):
+            data.append(this_note.split('-')[1].strip())
+        elif this_note.startswith('IP'):
+            data.append(this_note.split('-')[1].strip())
```

Comment:
Well, for no good reason....  I've made it one clause.  I also pulled that out to a method for marginal reasons (this could get more sophisticated in what it searches, so seemed worthwhile to just pull out now).